### ISSUE TITLE
#2110 has broken el.class() removing styles

### ISSUE THREAD

**Issue by ohemsted-thales** [cytoscape/cytoscape.js#2336]
**Issue type**
Bug report

**Environment info**

- Cytoscape.js version : 3.5.0
- Browser/Node.js & version : Chrome 72.0.3626.121 (Official Build) (64-bit)

**Current (buggy) behaviour**
Previously when calling:
`this.chart.elements().classes();`
this would clear all classes on all elements (as per the example http://js.cytoscape.org/#eles.classes)

**Desired behaviour**
All classes should be cleared as per the example. (http://js.cytoscape.org/#eles.classes)

**Minimum steps to reproduce**
```js
this.chart.elements().addClass('testClass');
this.chart.elements().classes(); // Expect testClass to have been removed from all elements
```

**Comment by maxkfranz** [comment]
I think the documentation just has a typo.  At some point (way back) I converted parts of the documentation to use single quotes instead of double quotes and somehow this example became malformed.  The example used to be --- and should be --- `ele.classes('')` (empty string).  

Unfortunately, this seems to be a mistake in the docs that just happened to work.  Sorry for the confusion.

**Comment by ohemsted-thales** [comment]
We had previously used `this.chart.elements().classes();` to clear all styles. This looks like a change in behaviour rather than a documentation issue. 

I wasn't sure if this change in behaviour was intended, it was broken as part of a minor release.

**Comment by maxkfranz** [comment]
This is a case of the documentation having a typo and a bug that makes the typo work.  It's explained a bit in #2341.

I looked into the details a bit more.  Basically, there was the typo in the documentation and someone complained about it.  He made the same assumption as you: `ele.classes(undefined)` should remove classes, because that's what the example says.  At that point in time, `ele.classes(undefined)` didn't mutate anything --- the correct behaviour.

I made the same assumption as that person did.  I took the example in the documentation as gospel even though it was just a typo.  I shouldn't have --- but I did.  I don't remember exactly why.  Maybe I had too much faith in the documentation, so I followed it blindly.  Or maybe I just didn't have enough coffee that day.  In any case, I made the mistake of making the library behave like the typo to resolve the issue.  Instead of fixing the real issue, I made it worse by introducing a bug that matched the typo.

I'm not going to repeat the same mistake now.  There isn't a perfect solution, but I think this is the best path forward:  There will be a warning in v3.4.4 to make it easier for consumers of the library to recover from the mistake and upgrade to v3.5, and v3.4.4 will continue to work in this two-wrongs-make-a-right mode so as not to disturb affected legacy projects.

I'm sorry for the confusion.
